contentcuration/contentcuration/frontend/channelEdit/components/edit/EditModal.vue

```diff
@@ -498,18 +500,68 @@
               .slice(0, -1)
               .join('.');
           }
-          this.createNode(
-            FormatPresets.has(file.preset) && FormatPresets.get(file.preset).kind_id,
-            { title, ...file.metadata }
-          ).then(newNodeId => {
-            if (index === 0) {
-              this.selected = [newNodeId];
-            }
-            this.updateFile({
-              ...file,
-              contentnode: newNodeId,
+          if (file.metadata.folders === undefined) {
+            this.createNode(
+              FormatPresets.has(file.preset) && FormatPresets.get(file.preset).kind_id,
+              { title, ...file.metadata }
+            ).then(newNodeId => {
+              if (index === 0) {
+                this.selected = [newNodeId];
+              }
+              this.updateFile({
+                ...file,
+                contentnode: newNodeId,
+              });
             });
-          });
+          } else if (file.metadata.folders) {
+            this.createNode('topic', file.metadata).then(newNodeId => {
+              file.metadata.folders.forEach(org => {
+                this.createNode('topic', org, newNodeId).then(topicNodeId => {
+                  org.files.forEach(orgFile => {
+                    const extra_fields = {};
+                    extra_fields['options'] = { entry: orgFile.resourceHref };
+                    extra_fields['title'] = orgFile.title;
+                    let file_kind = null;
+                    FormatPresetsList.forEach(p => {
+                      if (p.id === file.metadata.preset) {
+                        file_kind = p.kind_id;
+                      }
+                    });
+
+                    this.createNode(file_kind, extra_fields, topicNodeId).then(resourceNodeId => {
+                      return File.uploadUrl({
+                        checksum: file.checksum,
+                        size: file.file_size,
+                        type: 'application/zip',
```

Comment:
Let's double check if we need to specify this explicitly, or if it should be inferred by existing functionality.